### DISCUSSION TITLE
How Can I Add !important To Classes

### DISCUSSION THREAD

**Discussion by khaledOghli** [unocss/unocss#2135] [Q&A] [answered]
Hi Guys, 
How Can I Add !important To all unocss Classes !?

**Comment by zyyv** [comment, marked as answer]
Use `postprocess` hook. 
[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIAuBTAHpgWgCdsQod8CAmXOKAMwHcCA2ABjfQD4AoKKAPYA7AiGwBbBELIB2qGCFJMUzFDn0ArnDgBPWBqUDxYAF4IARnGw9gAenAQuQA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiQYAHpFjpsAQy4AbeJhz4iJYKQAUyVFDO9ABccADaVKhhehJcdDA6ALSBThh65IhwUk7QoXq0aHpwagC6VCWydAAW0DCxMHSh-qj6sfFJ1bX1enkwVjCJTqRwfZYD0Pa4pBhwNQBuGFDBowN99k5mqGqyIgqN4ZFyokoqPgCUssLyjCxsnDx8-OeXRwqS0n6HqHQS6xihAEYAHQAJheqAkaFweSqrDAjQA9Aj6CAgdUEZsWmoLuUdhB4jQIBJ6HQfBAOAArM7NVDkilAjC4O70IFYaAAUXsEiqPmAAF4AHxwYBhAElOB84Wi8UAan0cAAhKAbDBJjA9GdDlRtkhsUggA&options=N4IgzgLgTglgxhEAuaBXApgGhNAhgOzADMB7KAW2SNwBsx0BfIA)

**Comment by christhofer** [comment]
In TailwindCSS, we can use [Selector Strategy](https://tailwindcss.com/docs/configuration#selector-strategy) instead of adding `!important` anywhere.

```js
// tailwind.config.js
module.exports = {
    // ...
    important: '#app',
}
```

How to do this with postprocess? Basically change this
```css
.btn {}
.form-input {}
```

into
```css
#app .btn {}
#app .form-input {}
```

For my use case, I'm embedding a Vue app inside an HTML block on WordPress builder, I want to make sure my styles from Vue don't affect WordPress style, and vice versa.

**Reply by christhofer** [reply]
NVM, I get it to work after figuring out the `obj` argument contains property `selector`

[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIAuBTAHpgWgCdsQod8CAmXOKAMwHcCA2ABjfQD4AoKKAPYA7AiGwBbBELIB2qGCFJMUzFDn0ArnDgBPWBqUDxYAF4IARnGw9gAenAQuQA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAqrhJQL5xZQQhwDkAV1YBjOnX5IkGAB6RY6bAENBAG3iYc%2BIiWCkAFMlR0AFtBgjBMOgC5EVVP0t0YvALSnzlmPzv8YsjBuqqRwATJB0Eq4pBhwZgBuGFA24UEBSqqSqOyUqDT0jLZwANoO1LQMzKz6AJR5cAC6DZAuNBAi9HT6EABGAFa1RqhwIsQucAyqGCKuUHAAvAIAxEpgYNkjff0AdBi4MFDA9DtY0ACiSiIm%2BsALAHz2IyPAWHD6AITbO1Mzcz8wJSwOgAdWAMBuAAMACQIX6zaCcSG1WpPZ5bAY-DDTBHzJYwuHYv6IuCQuAAajg33hc3KOXK7FqVFySEZSCAA&css=Q&options=N4IgzgLgTglgxhEAuaBXApgGhNAhgOzADMB7KAW2SNwBsx0BfIA)
